File: 洛谷题单/入门5_字符串/P1957_口算练习题/solution.py
n = int(input())
op = ''

for _ in range(n):
    parts = input().split()
    
    if len(parts) == 3:
        if parts[0] == 'a': op = '+'
        elif parts[0] == 'b': op = '-'
        else: op = '*'
        a, b = int(parts[1]), int(parts[2])
    else:
        a, b = int(parts[0]), int(parts[1])
    
    if op == '+': result = a + b
    elif op == '-': result = a - b
    else: result = a * b
    
    expr = f"{a}{op}{b}={result}"
    print(expr)
    print(len(expr))

File: 洛谷题单/入门5_字符串/P1957_口算练习题/solution.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        
        while (T-- > 0) {
            String s = sc.next();
            String[] parts = s.split("[+=]");
            long a = Long.parseLong(parts[0]);
            long b = Long.parseLong(parts[1]);
            long c = Long.parseLong(parts[2]);
            
            System.out.println(a + b == c ? "YES" : "NO");
        }
    }
}
